tax/errors.go

```diff
@@ -0,0 +1,30 @@
+package tax
+
+import "fmt"
+
+// Error is a general wrapper around tax errors produced during run
+// time, typically during calculations. Not to be confused with errors
+// produced from definition validation.
+type Error struct {
+	Reason Key
+}
+
+// Standard list of tax errors
+var (
+	ErrInvalidCategory      = &Error{Reason: "invalid-category"}
+	ErrInvalidRate          = &Error{Reason: "invalid-rate"}
+	ErrInvalidDate          = &Error{Reason: "invalid-date"}
+	ErrInvalidPricesInclude = &Error{Reason: "invalid-prices-include"}
```

Comment:
Consider using constants, so that these package variables can't be changed at runtime.

Or unexport the `Error` type, then they can't be modified by any code outside of this package.

Reply:
I ended up refactoring this hence why it looks like this. Yes, that would make sense.